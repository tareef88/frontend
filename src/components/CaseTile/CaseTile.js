import React from 'react';
import useStyles from './styles';

function getDateFormat(timestamp) {
  const date = new Date(timestamp * 1000);
  const year = date.getFullYear();
  const month = date.getMonth();
  const day = date.getDay();
  const formattedTime = `${year}-${month}-${day}`;
  return formattedTime;
}

function CaseTile({ singleCase }) {
  const classes = useStyles();
  return (
    <div
      key={singleCase.id}
      className={classes.main}
    >
      <div className={classes.imageContainer}>
        <img className={classes.tileImage} alt="Bike" src={singleCase.media.image_url} />
      </div>
      <div className={classes.infoContainer}>
        <div className={classes.titleDescriptionContainer}>
          <text className={classes.title}>{singleCase.title}</text>
          <text className={classes.description}>{singleCase.description}</text>
        </div>
        <div className={classes.dateAddressContainer}>
          <text className={classes.date} style={{ fontSize: 'calc(12px + 1vw)' }}>
            {getDateFormat(singleCase.occurred_at)}
            {' '}
            {singleCase.address}
          </text>
        </div>
      </div>

    </div>
  );
}
export default CaseTile;
