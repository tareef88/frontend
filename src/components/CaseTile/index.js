import CaseTile from './CaseTile';

export default CaseTile;
