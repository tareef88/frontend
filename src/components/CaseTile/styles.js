import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  main: {
    display: 'flex',
    flexDirection: 'row',
    height: 250,
    marginTop: 30,
    borderRadius: 15,
    overflow: 'hidden',
    '-webkit-box-shadow': '0px 7px 40px 6px rgba(0,0,0,0.75);',
    '-moz-box-shadow': '0px 7px 40px 6px rgba(0,0,0,0.75);',
    'box-shadow': '0px 7px 40px 6px rgba(0,0,0,0.75);',
  },
  imageContainer: {
    width: '25%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
  },
  infoContainer: {
    textAlign: 'left',
    width: '75%',
    display: 'flex',
    flexDirection: 'column',
    padding: 10,
  },
  title: {
    fontSize: 14,
    fontWeight: '900',
  },
  description: {
    fontSize: 14,
    marginTop: 10,
  },
  date: {
    fontSize: 14,
  },
  address: {
    fontSize: 14,

  },
  tileImage: {
    width: 90,
    height: 80,
  },
  titleDescriptionContainer: {
    display: 'flex',
    flexDirection: 'column',
    height: '70%',
    paddingLeft: 10,
  },
  dateAddressContainer: {
    height: '30%',
    display: 'flex',
    flexDirection: 'row',
    paddingLeft: 10,
    alignItems: 'center',
  },
  '@media (min-width: 760px)': {
    tileImage: {
      width: 180,
      height: 150,
    },
    main: {
      height: 300,
    },
    title: {
      fontSize: 16,
    },
    description: {
      fontSize: 16,
    },
    date: {
      fontSize: 16,
    },
    address: {
      fontSize: 16,
    },
  },
  '@media (min-width: 1000px)': {
    tileImage: {
      width: 250,
      height: 200,
    },
    title: {
      fontSize: 20,
    },
    description: {
      fontSize: 20,
    },
    date: {
      fontSize: 20,
    },
    address: {
      fontSize: 20,
    },
    main: {
      height: 350,
    },
  },

});

export default useStyles;
