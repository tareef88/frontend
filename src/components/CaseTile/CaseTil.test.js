import React from 'react';
import renderer from 'react-test-renderer';
import CaseTile from './CaseTile';

const singleCase = {
  id: 97092,
  title: 'Stolen 2017 Raleigh Tourist de Luxe(green)',
  description: '',
  address: 'Frederiksberg, 2000, DK',
  occurred_at: 1550851200,
  updated_at: 1568736467,
  url: 'https://bikewise.org/api/v1/incidents/97092',
  source: {
    name: 'BikeIndex.org',
    html_url: 'https://bikeindex.org/bikes/581373',
    api_url: 'https://bikeindex.org/api/v1/bikes/581373',
  },
  media: {
    image_url: null,
    image_url_thumb: null,
  },
  location_type: null,
  location_description: null,
  type: 'Theft',
  type_properties: null,
};

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <CaseTile singleCase={singleCase} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
