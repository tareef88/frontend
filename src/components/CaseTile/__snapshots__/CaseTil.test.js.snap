// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="main-0-2-1"
>
  <div
    className="imageContainer-0-2-2"
  >
    <img
      alt="Bike"
      className="tileImage-0-2-8"
      src={null}
    />
  </div>
  <div
    className="infoContainer-0-2-3"
  >
    <div
      className="titleDescriptionContainer-0-2-9"
    >
      <text
        className="title-0-2-4"
      >
        Stolen 2017 Raleigh Tourist de Luxe(green)
      </text>
      <text
        className="description-0-2-5"
      >
        
      </text>
    </div>
    <div
      className="dateAddressContainer-0-2-10"
    >
      <text
        className="date-0-2-6"
        style={
          Object {
            "fontSize": "calc(12px + 1vw)",
          }
        }
      >
        2019-1-5
         
        Frederiksberg, 2000, DK
      </text>
    </div>
  </div>
</div>
`;
